src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/MD5CryptoServiceProvider.cs

```diff
@@ -19,8 +20,15 @@ public MD5CryptoServiceProvider()
 
         public override void Initialize()
         {
-            // Nothing to do here. We expect HashAlgorithm to invoke HashFinal() and Initialize() as a pair. This reflects the
-            // reality that our native crypto providers (e.g. CNG) expose hash finalization and object reinitialization as an atomic operation.
+            Span<byte> destination = stackalloc byte[HashSizeBits / 8];
+
+            if (!_incrementalHash.TryGetHashAndReset(destination, out _))
+            {
+                Debug.Fail("Reset expected a properly sized buffer.");
+                throw new CryptographicException();
+            }
+
+            CryptographicOperations.ZeroMemory(destination);
```

Comment:
I don't think I'd bother zeroing out discarded hash outputs.  Happy to hear counter-arguments, but this doesn't hit my "sensitive enough" trigger (contrast to the temporary buffer used for RSA decryption, where a private key had to be used).

Reply:
Yeah, I was iffy on it, too. I have no strong arguments.